addons/purchase/purchase.py

```diff
@@ -1018,7 +1017,11 @@ def onchange_product_id(self, cr, uid, ids, pricelist_id, product_id, qty, uom_i
             # The 'or not uom_id' part of the above condition can be removed in master. See commit message of the rev. introducing this line.
             dummy, name = product_product.name_get(cr, uid, product_id, context=context_partner)[0]
             if product.description_purchase:
-                name += '\n' + product.description_purchase
+                desciption_purchase_label = ir_translation._get_source(
```

Comment:
Sorry @pedrobaeza I don't get what you mean, if you look at [line 1014](https://github.com/OCA/OCB/pull/238/files#diff-fe8587929089e5fdb6c75687011b1c7eR1014) the product object is already browsed using context with partner lang key and is not getting the correct translation, that's why we invoke the actual translation here, something similar to what is being doing [here](https://github.com/OCA/OCB/blob/7.0/addons/account/account_invoice.py#L670)